### PR TITLE
Updates for immutable List mutations

### PR DESCRIPTION
**Pull request by TonyRice**
Not sure if we should just go with a shallow copy for this, or if a deep copy is needed for the list.

### CODE REVIEW

**Comment by wilzbach** [comment]
Deep copy is required. Storyscript doesn't have head-const immutability, but full transitive immutability.

File: storyruntime/processing/mutations/ListMutations.py [hunk 1]
```diff
@@ -19,24 +19,32 @@ def length(cls, mutation, value, story, line, operator):
     @classmethod
     def append(cls, mutation, value, story, line, operator):
         item = story.argument_by_name(mutation, 'item')
-        return value.append(item)
+        copy = value.copy()
```
Comment (wilzbach) [inline storyruntime/processing/mutations/ListMutations.py]:
```suggestion
        copy = value.deepcopy()
```

**Comment by judepereira** [comment]
Deep copying from the root call at https://github.com/storyscript/runtime/blob/6d50276314791d8e066d4b1ce505483f1668d37d/storyruntime/processing/Mutations.py#L13. Will open up a new PR.

**Comment by TonyRice** [comment]
> Deep copying from the root call at
> 
> https://github.com/storyscript/runtime/blob/6d50276314791d8e066d4b1ce505483f1668d37d/storyruntime/processing/Mutations.py#L13
> 
> . Will open up a new PR.

We should probably communicate with each other directly when we're making decisions to invalidate someone's code without any clear reason. For example: "I think this solution might also work because of XXX"

The solution in: https://github.com/storyscript/runtime/pull/547 definitely works great, btw.

**Comment by judepereira** [comment]
Got it, will do. Thanks @TonyRice!

**Pull request closed without merging**